src/lib/ndpi_utils.c

```diff
@@ -846,6 +846,21 @@ int ndpi_has_human_readeable_string(struct ndpi_detection_module_struct *ndpi_st
 
 /* ********************************** */
 
+const char* ndpi_get_flow_info(struct ndpi_flow_struct const * const flow,
+                               ndpi_protocol const * const l7_protocol)
+{
+  switch (l7_protocol->master_protocol)
+  {
+    case NDPI_PROTOCOL_DNS: return (char *)flow->host_server_name;
+    case NDPI_PROTOCOL_HTTP: return (char *)flow->host_server_name;
+    case NDPI_PROTOCOL_TLS: return flow->protos.tls_quic_stun.tls_quic.client_requested_server_name;
```

Comment:
Even if not explicitly required, might it be useful to handle QUIC case, too?

Reply:
Agreed, even it was not explicitly mentioned in #1253.